docs/docs/unit-testing.md

```diff
@@ -0,0 +1,257 @@
+                                                                                            ^^^^^^
+SyntaxError: Unexpected token import
+```
+
+This is because `gatsby-browser-entry.js` isn't being transpiled before running
+in Jest. You can fix this by changing the default `transformIgnorePatterns` to
+exclude the `gatsby` module.
+
+## Writing tests
+
+A full guide to unit testing is beyond the scope of this guide, but you can
+start with a simple snapshot test to check that everything is working.
+
+First, create the test file. You can either put these in a `__tests__`
+directory, or put them elsewhere (usually next to the component itself), with
+the extension `.spec.js` or `.test.js`. The decision comes down to your own
+taste. For this guide you will be testing the `<Bio />` component, so create a
+`Bio.test.js` file next to it in `src/components`:
+
+```js
+import React from "react"
+
+import renderer from "react-test-renderer"
+
+import Bio from "./Bio"
+
+describe("Bio", () =>
+  it("renders correctly", () => {
+    const tree = renderer.create(<Bio />).toJSON()
+    expect(tree).toMatchSnapshot()
+  }))
+```
+
+This is a very simple snapshot test, which uses `react-test-renderer` to render
+the component, and then generates a snapshot of it on first run. It then
+compares future snapshots against this, which means you can quickly check for
+regressions. Visit [the Jest docs](https://jestjs.io/docs/en/getting-started) to
+learn more about other tests that you can write.
+
+## Running tests
+
+If you look inside `package.json` you will probably find that there is already a
+script for `test`, which just outputs an error message. Change this to simply
+`jest`:
+
+```json
+  "scripts": {
+    "test": "jest"
+  }
+```
+
+This means you can now run tests by typing `npm test`. If you want you could
+also add a script that runs `jest --watchAll` to watch files and run tests when
+they are changed.
+
+Now, run `npm test` and you should immediately get an error like this:
+
+```sh
+ @font-face {
+    ^
+
+    SyntaxError: Invalid or unexpected token
+
+      2 |
+      3 | // Import typefaces
+    > 4 | import 'typeface-montserrat'
+```
+
+This is because the CSS mock doesn't recognize the `typeface-` modules. You can
+fix this easily by creating a new manual mock. Back in the `__mocks__`
+directory, create a file called `typeface-montserrat.js` and another called
+`typeface-merriweather.js`, each with the content `{}`. Any file in the mocks
+folder which has a name that matches that of a node_module is automatically used
+as a mock.
+
+Run the tests again now and it should all work! You should get a message about
+the snapshot being written. This is created in a `__snapshots__` directory next
+to your tests. If you take a look at it, you will see that it is a JSON
+representation of the `<Bio />` component. You should check your snapshot files
+into your SCM repository so that anyone can see if the tests have stopped
```

Comment:
Wording.

> so that any changes are tracked in history

Reply:
And that tests can run by CI solutions like travis (which could have it's own guide that could be linked in the future)